essential-projects/iam
# add config for jwt adapter, add RSA support, add expiration support

## What did you change?

- The shared secret is included in code, this has to be removed (in case there is no appropriate config, there is a fallback for backward compatibility).
- The algorithm used to sign the JWT should be configurable (default is HS256).
- The shared secret (HS256) or the RSA key (RS256) should be configurable.
- The lease time of the JWT should be configurable.

> Fixes # (issue, if applicable)
https://github.com/essential-projects/iam/issues/4

## How can others test the changes?

1. add a config for `iam:jwt_token_adapter`. Example:
```
{
  "algorithm": "RS256",
  "key": "demo_master",
  "expiresIn": "8h"
}
``` 
or 
```
{
  "algorithm": "HS256",
  "secret": "secret",
  "expiresIn": "8h"
}
``` 
Description of valid algorithms or expiration settings can be found here [github.com/auth0/node-jsonwebtoken](https://github.com/auth0/node-jsonwebtoken).

2. run integration tests or app and try to login. A valid JWT should be returned.

## PR-Checklist

Please check the boxes in this list after submitting your PR:
